packages/diracx-web-components/components/shared/FilterForm.tsx

```diff
@@ -116,34 +139,155 @@ export function FilterForm(props: FilterFormProps) {
+                        ? tempFilter.values
+                        : tempFilter.value
+                    }
+                    onChange={(e) =>
+                      onChange(
+                        ["in", "not in"].includes(tempFilter.operator)
+                          ? "values"
+                          : "value",
+                        e.target.value,
+                      )
+                    }
+                    multiple={["in", "not in"].includes(tempFilter.operator)}
+                    sx={{ minWidth: 100 }}
+                  >
+                    {selectedColumn?.type.map((val: string) => (
+                      <MenuItem key={val} value={val}>
+                        {selectedColumn.render
+                          ? selectedColumn.render(val)
+                          : val}
+                      </MenuItem>
+                    ))}
+                  </Select>
+                </>
+              ) : selectedColumn?.type == "DateTime" &&
+                tempFilter.operator != "last" ? (
+                <DateTimePicker
+                  label="Value"
+                  value={dayjs(tempFilter.value)}
+                  onChange={(e) => onChange("value", e?.toISOString() || "")}
+                  views={["year", "day", "hours", "minutes", "seconds"]}
+                />
+              ) : selectedColumn?.type == "DateTime" &&
+                tempFilter.operator == "last" ? (
+                <>
+                  <InputLabel id="value">Value</InputLabel>
+                  <Select
+                    label="Value"
+                    labelId="value"
+                    value={tempFilter.value}
+                    onChange={(e) => onChange("value", e.target.value)}
+                    sx={{ minWidth: 100 }}
+                  >
+                    <MenuItem value="hour">Hour</MenuItem>
+                    <MenuItem value="week">Week</MenuItem>
+                    <MenuItem value="month">Month</MenuItem>
+                    <MenuItem value="year">Year</MenuItem>
+                  </Select>
+                </>
+              ) : selectedColumn?.type == "number" &&
+                !["in", "not in", "like"].includes(tempFilter.operator) ? (
+                <TextField
+                  id="value"
+                  variant="outlined"
+                  label="Value"
+                  value={tempFilter.value}
+                  onChange={(e) => onChange("value", e.target.value)}
+                  type="number"
+                />
+              ) : selectedColumn?.type == "number" &&
+                ["in", "not in"].includes(tempFilter.operator) ? (
+                <TextField
+                  id="value"
+                  variant="outlined"
+                  label="Value"
+                  value={tempFilter.values?.join(" ")}
+                  onChange={(e) =>
+                    onChange("values", e.target.value.split(" "))
+                  }
+                />
+              ) : (
```

Comment:
Same kind of comment here, I wonder if part of the logic could go in a structure with each type of filter. To be investigated (we would have a `strategy` pattern).